pandas/tests/indexing/interval/test_interval.py

```diff
@@ -222,3 +222,15 @@ def test_reindex_behavior_with_interval_index(self, base):
         expected_result = Series([np.nan, 0], index=[np.nan, 1.0], dtype=float)
         result = ser.reindex(index=[np.nan, 1.0])
         tm.assert_series_equal(result, expected_result)
+
+    def test_multiindex_with_interval_index(self):
+        # for GH#25298
+        intIndex = IntervalIndex.from_arrays([1, 5, 8, 13, 16], [4, 9, 12, 17, 20])
+        multiIndex = pd.MultiIndex.from_arrays([["a", "a", "b", "b", "c"], intIndex])
+        data = [(1, 2), (3, 4), (5, 6), (7, 8), (9, 10)]
+        df = DataFrame(data, index=multiIndex)
+        result1 = df.loc[("b", 16)]
+        expected = Series([7, 8])
+        tm.assert_series_equal(result1, expected, check_names=False)
+        result2 = df.loc["b"].loc[16]
```

Comment:
Thanks, @harinik, it looks good. Just one question: do we need the second check (result2)? I think this example already worked as expected.